Add documentation for adding new repositories

lluiscampos opened the pull request.

lluiscampos commented:
Take it as a first draft. I wrote it pretty fast as I got considerably tired... Thanks for your feedback!

alfrunes commented on Documentation/adding-new-repository.md:
```diff
@@ -0,0 +1,111 @@
+
+## GitLab: project
+
+To craete the project in GitLab:
```
```suggestion
To create the project in GitLab:
```

alfrunes commented on Documentation/adding-new-repository.md:
```diff
@@ -0,0 +1,111 @@
+
+1. Open an incognito window in your browser
+2. Log into GitHub as `mender-test-bot`, credentials are in `mystiko`
+3. Log into GitLab, using OAth from GitHub
```
```suggestion
3. Log into GitLab, using OAuth from GitHub
```

lluiscampos commented:
The rendered markdown file can be seen [here](https://github.com/lluiscampos/mender-qa/blob/Documentation/adding-new-repository.md/Documentation/adding-new-repository.md).

alfrunes commented on Documentation/adding-new-repository.md:
```diff
@@ -0,0 +1,111 @@
+
+### Enterprise fork
+
+TODO
```
Don't forget to fill in this section :)

lluiscampos replied:
Actually, I don't really know what is special about it (the only thing I remember is for `integration-test-bot`, and I added it in that section already). I was thinking to leave it TODO until we do a new repository with enterprise fork.

Or do you know what else is special about repos with enterprise fork?

alfrunes replied:
I don't know anything on top of my head either. I just pointed it out because it's easy to forget about following up inline `TODO`s (at least for me).
Perhaps you can remove this subsection since it's already covered?

kacf replied:
There are a couple of things I can think of:

1. Change the license to All Rights Reserved
2. Implement closed source license test for Golang files (`CheckMenderCompliance`).

alfrunes replied:
And maybe also mention that enterprise Docker repositories are published exclusively to `registry.mender.io`.

lluiscampos replied:
Thanks. Added

alfrunes commented on Documentation/adding-new-repository.md:
```diff
@@ -0,0 +1,111 @@
+
+## integration
+
+*NOTE*: needs to be done togeher with `mender-qa`.
```
```suggestion
*NOTE*: needs to be done together with [mender-qa](mender-qa).
```

alfrunes reviewed (changes requested):
Nice work, I have a couple of corrections for some typos I came across.

merlin-northern reviewed (approved):
this is really nice and much needed! thanks a lot!
one thing that I would love to see here is the case of "GitLab master branch not synced with GitHub", how to detect and what to do then.

oleorhagen commented on Documentation/adding-new-repository.md:
```diff
@@ -0,0 +1,111 @@
+  * ref https://github.com/mendersoftware/mender-qa/blob/master/scripts/jenkins-yoctobuild-build.sh
+  * Add the new Docker images, if applicable, to `Build server repositories.`
+
+## integration-test-runner
```
Question:
Maybe we should specify that `integration-test-runner = mender-test-bot`?

lluiscampos replied:
@oleorhagen Not sure. `mender-test-bot` is our GitHub user (unrelated, if you ask me), `integration-test-runner` is the repository and the Docker image name, `test-runner-mender-io` is the deployment name in Kubernetes... From all these, I think the repository name is the best reference.

Can you give me an example on how would you phrase it?

oleorhagen replied:
It's minor, my thinking was just:
Are everyone aware that the `integration-test-runner` is actually what is running the `mender-test-bot` account on Github?

lluiscampos replied:
Okey, fixed.

oleorhagen reviewed (approved):
Real nice!

kacf commented on Documentation/adding-new-repository.md:
```diff
@@ -0,0 +1,111 @@
+  * ref https://github.com/mendersoftware/mender-qa/blob/master/scripts/jenkins-yoctobuild-build.sh
+  * Add the new Docker images, if applicable, to `Build server repositories.`
+
+## integration-test-runner
```
One section you might want to add after this is:

```
## Docker images

Add Docker project on Docker hub, with correct permissions, etc, etc
```

lluiscampos replied:
Very good catch, thanks.

lluiscampos replied:
(I will add it before `GitLab: CI/CD pipeline`, as the image repo needs to be in place before the pipeline)

lluiscampos replied:
I added:

```
## Docker image(s)

When applicable, prepare Docker image repositories for the new component.

* For Open Source repository, create a new public repository in Docker Hub. Teams and permissions:
  * admin: Admin
  * developers: Read-only
  * openrnd: Read-only
  * robots: Read & Write
* For closed source repository, create a new private repository in registry.mender.io
```

lluiscampos commented:
> Nice work, I have a couple of corrections for some typos I came across.

Thanks for the typos. Fixed.

> There are a couple of things I can think of:
> 
>     1. Change the license to All Rights Reserved
> 
>     2. Implement closed source license test for Golang files (`CheckMenderCompliance`).

Fixed.

> One section you might want to add after this is:
> 
> ```
> ## Docker images
> 
> Add Docker project on Docker hub, with correct permissions, etc, etc
> ```

Fixed.

> Question:
> Maybe we should specify that `integration-test-runner = mender-test-bot`?

Fixed.

> this is really nice and much needed! thanks a lot!
> one thing that I would love to see here is the case of "GitLab master branch not synced with GitHub", how to detect and what to do then.

@merlin-northern I don't know where would be a good place for this. Our [Knowledge Base](https://gitlab.com/Northern.tech/Mender/knowledge-base/-/wikis/home) maybe?

alfrunes reviewed (approved):
LGTM
